core/job_generator.go

```diff
@@ -0,0 +1,108 @@
+package core
```

Comment:
Filename `core/job_generator.go` confused me a bit because there is both generator and producer. How about job_batch.go or sth?